mParticle-Apple-SDK/Data Model/MPConsumerInfo.mm

```diff
@@ -176,39 +176,54 @@ - (NSDictionary *)dictionaryRepresentation {
 @end
 
 #pragma mark - MPConsumerInfo
+@interface MPConsumerInfo ()
+@property (nonatomic, strong, nonnull) dispatch_queue_t queue;
+@end
+
 @implementation MPConsumerInfo
 
 @synthesize cookies = _cookies;
 @synthesize mpId = _mpId;
 @synthesize uniqueIdentifier = _uniqueIdentifier;
 
 - (id)init {
+    // Serial since read/write lock not appropriate given how mpId is created.
+    dispatch_queue_t newQueue = dispatch_queue_create("com.mParticle.MPCookie.queue", DISPATCH_QUEUE_SERIAL);
+    
     self = [super init];
-    if (self) {
+    if (self && newQueue) {
+        _queue = newQueue;
         _consumerInfoId = 0;
+    } else {
+        self = nil;
     }
     
     return self;
 }
 
 #pragma mark NSCoding
 - (void)encodeWithCoder:(NSCoder *)coder {
-    if (self.cookies) {
-        [coder encodeObject:_cookies forKey:@"cookies"];
-    }
-    
-    if (self.mpId) {
-        [coder encodeObject:_mpId forKey:@"mpId"];
-    }
-    
-    if (self.uniqueIdentifier) {
-        [coder encodeObject:_uniqueIdentifier forKey:@"uniqueIdentifier"];
-    }
+    dispatch_sync(self.queue, ^{
+        if (self.cookies) {
+            [coder encodeObject:_cookies forKey:@"cookies"];
+        }
+        
+        if (self.mpId) {
+            [coder encodeObject:_mpId forKey:@"mpId"];
+        }
+        
+        if (self.uniqueIdentifier) {
+            [coder encodeObject:_uniqueIdentifier forKey:@"uniqueIdentifier"];
+        }
+    });
 }
 
 - (id)initWithCoder:(NSCoder *)coder {
+    // Serial since read/write lock not appropriate given how mpId is created.
+    dispatch_queue_t newQueue = dispatch_queue_create("com.mParticle.MPCookie.queue", DISPATCH_QUEUE_SERIAL);
     self = [super init];
     if (self) {
+        self.queue = newQueue;
```

Comment:
If the implementation creates a queue lazily, we probably wouldn't need to create and assign one over here.